jujugui/static/gui/src/app/store/env/fakebackend.js

```diff
@@ -1592,28 +1600,28 @@ YUI.add('juju-env-fakebackend', function(Y) {
     },
 
     /**
-     * Utility to promise to load a charm for serviceData.
-     * @method _promiseCharmForService
-     * @param {Object} serviceData to load charm for. seviceData is the
-     *        imported service attributes, so .charm should be the charm
-     *        url.
-     * @return {Promise} resolving with [charm model, serviceData].
+     * Utility to promise to load a charm for applicationData.
+     * @method _promiseCharmForApplication
+     * @param {Object} applicationData to load charm for. applicationData is
+     *        the imported application attributes, so .charm should be the
+     *        charm url.
+     * @return {Promise} resolving with [charm model, applicationData].
      */
-    _promiseCharmForService: function(serviceData) {
+    _promiseCharmForApplication: function(applicationData) {
       var self = this,
-          charmId = serviceData.charm;
+          charmId = applicationData.charm;
 
       return Y.Promise(function(resolve, reject) {
```

Comment:
At some point in the future we should update these to use native promises so that we can get better stacktraces.